src/storage/engine/rocksdb.rs

```diff
@@ -18,17 +18,41 @@ use rocksdb::rocksdb::Snapshot as RocksSnapshot;
 use kvproto::kvrpcpb::Context;
 use storage::{Key, Value, KvPair};
 use util::hex;
-use super::{Engine, Snapshot, Modify, Result};
+use tempdir::TempDir;
+use super::{Engine, TEMP_DIR, Snapshot, Modify, Result};
 
+
+
+#[allow(dead_code)]
 pub struct EngineRocksdb {
     db: DB,
+    // only use for memory mode
+    temp_dir: Option<TempDir>,
 }
 
 impl EngineRocksdb {
     pub fn new(path: &str) -> Result<EngineRocksdb> {
```

Comment:
I don't think so, It is very strange to using an string for distinguishing here, and I can't use "temp-dir" outer if I really want to use it.